src/nyx/python/new_bindings_py.cpp

```diff
@@ -118,6 +118,72 @@ py::tuple process_data(
     return py::make_tuple(pyHeader, pyStrData, pyNumData);
 }
 
+py::tuple featurize_fname_lists_imp (const py::list& int_fnames, const py::list & seg_fnames)
+{
+    std::vector<std::string> intensFiles, labelFiles;
+    for (auto it = int_fnames.begin(); it != int_fnames.end(); ++it)
+    {
+        std::string fn = it->cast<std::string>();
+        intensFiles.push_back(fn);
+    }
+    for (auto it = seg_fnames.begin(); it != seg_fnames.end(); ++it)
+    {
+        std::string fn = it->cast<std::string>();
+        labelFiles.push_back(fn);
+    }
+
+    // Check the file names 
+    if (intensFiles.size() == 0)
+        throw std::runtime_error("Intensity file list is blank");
+    if (labelFiles.size() == 0)
+        throw std::runtime_error("Segmentation file list is blank");
+    if (intensFiles.size() != labelFiles.size())
+        throw std::runtime_error("Imbalanced intensity and segmentation file lists");
+    for (auto i = 0; i < intensFiles.size(); i++)
+    {
+        const std::string& i_fname = intensFiles[i];
+        const std::string& s_fname = labelFiles[i];
+
+        if (!directoryExists(i_fname))
+        {
+            std::cout << i_fname << "\n";
```

Comment:
Filename should be part of the run-time error string thrown or else the user won't see it in a jupyter notebook. Change text to "File does not exist: /path/to/file" 

Also, the function being called is `directoryExists`. If it works on both directories or files then it should be renamed to something more generic like `exists` or `existsOnFilesystem` or something else of your choosing.